engine/battle/move_effects/brick_break.asm

```diff
@@ -2,28 +2,24 @@ BattleCommand_brickbreak:
 	ldh a, [hBattleTurn]
 	and a
 	ld hl, wEnemyScreens
-	ld bc, wEnemyLightScreenCount
 	jr z, .got_screens
 	ld hl, wPlayerScreens
-	ld bc, wPlayerLightScreenCount
 .got_screens
-	bit SCREENS_LIGHT_SCREEN, [hl]
-	jr z, .light_screen_done
-	res SCREENS_LIGHT_SCREEN, [hl]
-	xor a
-	ld [bc], a
+    ld a, [hl]
+	and a
+	ret z
+	push af
+	and SCREENS_REFLECT
+	jr z, .reflect_done
 	push hl
-	push bc
-	ld hl, BrokeLightScreenText
+	ld hl, BrokeReflectText
 	call StdBattleTextBox
-	pop bc
 	pop hl
-.light_screen_done
-	inc bc
-	bit SCREENS_REFLECT, [hl]
+.reflect_done
+	pop af
+	and SCREENS_LIGHT_SCREEN
+	ld a, 0 ; zero flag needs to be preserved
```

Comment:
Does this need to alter `a`, or could it do `ld [hl], 0` instead of `ld a, 0 / ld [hl], a`?

Reply:
no it doesnt need to alter `a` at all here but i was doing my best game freak impersonation and overengineering it :V